Mods/src/PuppiMod.cc

```diff
@@ -186,7 +190,35 @@ PuppiMod::Process()
 
   auto* pv = vertexes->At(0);
 
-  auto* pfCandidates = GetObject<PFCandidateCol>(fPFCandidatesName);
+  PFCandidateArr *InvertedParticles = new PFCandidateArr;
+  InvertedParticles->SetName(fInvertedParticlesName);
+
+  auto* inPfCandidates = GetObject<PFCandidateCol>(fPFCandidatesName);
+  PFCandidateCol* pfCandidates = NULL;
+  PFCandidateArr* leptonCandidates = NULL;
+
+  if (fNoLepton) {                                                                              // This is the default behavior of Puppi
+    leptonCandidates = new PFCandidateArr(16);
+    PFCandidateArr* tempCandidates = new PFCandidateArr(16);
```

Comment:
The same goes for tempCandidates - you don't want to be allocating & copying your input PF candidates here. How about

```
PFCandidateOArr leptonCandidates;
PFCandidateOArr noLeptonCandidates;
if (fNoLepton) {
  //  Add cand to leptonCandidates and noLeptonCandidates
  pfCandidates = &noLeptonCandidates;
}
else {
  pfCandidates = inPfCandidates;
}
```